src/main/java/org/dependencytrack/resources/v1/ConfigPropertyResource.java

```diff
@@ -156,5 +157,25 @@ public Response updateConfigProperty(List<ConfigProperty> list) {
         return Response.ok(returnList).build();
     }
 
+    @GET
+    @Path("/welcomeMessage")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Returns if a welcome Message is given, and wich should be showed", description = "<p>Requires permission <strong>SYSTEM_CONFIGURATION</strong></p>")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "A list of isMessageWelcome and messageWelcome", content = @Content(array = @ArraySchema(schema = @Schema(implementation = ConfigProperty.class)))),
+            @ApiResponse(responseCode = "401", description = "Unauthorized")
+    })
+    @AuthenticationNotRequired
+    public Response getGreetingMessage() {
+        try (QueryManager qm = new QueryManager(getAlpineRequest())) {
+            ConfigProperty greetingMessage = qm.getConfigProperty("Message", "welcomeMessage");
+            ConfigProperty isGreetingMessage = qm.getConfigProperty("Message", "isWelcomeMessage");
+            qm.close();
+            List<ConfigProperty> combined = new ArrayList<ConfigProperty>();
+            combined.add(greetingMessage);
+            combined.add(isGreetingMessage);
+            return Response.ok(combined).build();
+        }
+    }
```

Comment:
I'm not particularly fond of adding new endpoints for every new property that doesn't require authentication, especially considering the language-related question you asked in Slack. I think it would "scale" better if we added a new `/public` endpoint that would only return `ConfigProperty` records that we explicitly mark as "public".

We did something along those lines in Hyades, to mark properties as read-only or read-writable:

https://github.com/DependencyTrack/hyades-apiserver/blob/eca7ab7c672a2b128c2bb6d729b0fb12d532e60d/src/main/java/org/dependencytrack/model/ConfigPropertyConstants.java#L160-L162 

https://github.com/DependencyTrack/hyades-apiserver/blob/eca7ab7c672a2b128c2bb6d729b0fb12d532e60d/src/main/java/org/dependencytrack/resources/v1/AbstractConfigPropertyResource.java#L59-L66

You could follow a similar approach to filter properties that can safely be returned for in the `/public` endpoint. The frontend could then call this endpoint to retrieve welcome message and language preferences (and potentially more in the future).

Reply:
I figured that as well, easiest way would be adding a Property to the constants that mark them optionnely as public, like that:
```
    BOM_VALIDATION_TAGS_EXCLUSIVE("artifact", "bom.validation.tags.exclusive", "[]", PropertyType.STRING, "JSON array of tags for which BOM validation shall NOT be performed"),
    WELCOME_MESSAGE("general", "welcome.message.html", "%20%3Chtml%3E%3Ch1%3EYour%20Welcome%20Message%3C%2Fh1%3E%3C%2Fhtml%3E", PropertyType.STRING, "Custom HTML Code that is displayed before login", true),
    IS_WELCOME_MESSAGE("general", "welcome.message.enabled", "false", PropertyType.BOOLEAN, "Bool that says wheter to show the welcome message or not", true);

    private final String groupName;
    private final String propertyName;
    private final String defaultPropertyValue;
    private final PropertyType propertyType;
    private final String description;
    private final Boolean isPublic;


	ConfigPropertyConstants(String groupName, String propertyName, String defaultPropertyValue, PropertyType propertyType, String description) {
        this.groupName = groupName;
        this.propertyName = propertyName;
        this.defaultPropertyValue = defaultPropertyValue;
        this.propertyType = propertyType;
        this.description = description;
        this.isPublic = false;
    }
    
    ConfigPropertyConstants(String groupName, String propertyName, String defaultPropertyValue, PropertyType propertyType, String description, Boolean isPublic) {
        this.groupName = groupName;
        this.propertyName = propertyName;
        this.defaultPropertyValue = defaultPropertyValue;
        this.propertyType = propertyType;
        this.description = description;
        this.isPublic = isPublic;
    }
```